mailbox/jpa/src/main/java/org/apache/james/mailbox/jpa/quota/JPAPerUserMaxQuotaManager.java

```diff
@@ -0,0 +1,129 @@
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.mailbox.jpa.quota;
+
+import javax.persistence.EntityManager;
+import javax.persistence.EntityManagerFactory;
+
+import org.apache.james.mailbox.exception.MailboxException;
+import org.apache.james.mailbox.jpa.quota.model.MaxDefaultMessageCount;
+import org.apache.james.mailbox.jpa.quota.model.MaxDefaultStorage;
+import org.apache.james.mailbox.jpa.quota.model.MaxUserMessageCount;
+import org.apache.james.mailbox.jpa.quota.model.MaxUserStorage;
+import org.apache.james.mailbox.model.Quota;
+import org.apache.james.mailbox.model.QuotaRoot;
+import org.apache.james.mailbox.quota.MaxQuotaManager;
+
+import com.google.common.base.Function;
+import com.google.common.base.Optional;
+
+
+public class JPAPerUserMaxQuotaManager implements MaxQuotaManager {
+
+    private final EntityManager entityManager;
+
+    public JPAPerUserMaxQuotaManager(EntityManagerFactory entityManagerFactory) {
+        this.entityManager = entityManagerFactory.createEntityManager();
+    }
+
+    @Override
+    public void setMaxStorage(QuotaRoot quotaRoot, long maxStorageQuota) throws MailboxException {
+        entityManager.getTransaction().begin();
+        entityManager.merge(new MaxUserStorage(quotaRoot.getValue(), maxStorageQuota));
+        entityManager.getTransaction().commit();
+    }
+
+    @Override
+    public void setMaxMessage(QuotaRoot quotaRoot, long maxMessageCount) throws MailboxException {
+        entityManager.getTransaction().begin();
+        entityManager.merge(new MaxUserMessageCount(quotaRoot.getValue(), maxMessageCount));
+        entityManager.getTransaction().commit();
+    }
+
+    @Override
+    public void setDefaultMaxStorage(long defaultMaxStorage) throws MailboxException {
+        entityManager.getTransaction().begin();
+        entityManager.merge(new MaxDefaultStorage(defaultMaxStorage));
+        entityManager.getTransaction().commit();
+    }
+
+    @Override
+    public void setDefaultMaxMessage(long defaultMaxMessageCount) throws MailboxException {
+        entityManager.getTransaction().begin();
+        entityManager.merge(new MaxDefaultMessageCount(defaultMaxMessageCount));
+        entityManager.getTransaction().commit();
+    }
+
+    @Override
+    public long getDefaultMaxStorage() throws MailboxException {
+        MaxDefaultStorage storedValue = entityManager.find(MaxDefaultStorage.class, MaxDefaultStorage.DEFAULT_KEY);
+
+        return Optional.fromNullable(storedValue)
+            .transform(new Function<MaxDefaultStorage, Long>() {
+                @Override
+                public Long apply(MaxDefaultStorage input) {
+                    return input.getValue();
+                }
+            })
+            .or(Quota.UNLIMITED);
```

Comment:
They initially wrote this in java8 ^^

So I kept close to their code but I agree with you. I moved away of optional transformations.